### DISCUSSION TITLE
May an bug for test contract on testnets!

### DISCUSSION THREAD

**Discussion by JustinStar-py** [wevm/wagmi#3771] [General]
Yo community, im using wagmi in my react staking dapp and now i wanna test it, but when i click on stake button it doesnt send metamask tx window, but in mainnet it send me tx window.

error:
**TypeError: Cannot destructure property 'account' of 'parameters' as it is undefined.**

here code:
```
   const { writeContract, isSuccess: writeSuccess, isError: writeError, isPending: writePending } = useWriteContract()
   const { data: stakeData, status } = useSimulateContract({
        address: '0x027133f2521FBc163Fe7A95AB205384411084B12',
        abi: NYXEcosystemStakingABI,
        functionName: "stake",
        args: [10*10**18],
        stateMutability: 'nonpayable',
        chainId: 97 //bsc testnet
      });


    const handleInputClick = (input) => {
        setCurrentInput(input);
    }

    const handleStake = async () => {
        setTxPending(true);
        try {
            await writeContract(stakeData?.request)
        } catch (error) {
            setTxPending(false);
            setMessageInfo({ isLoading: false, messageText: `Error: ${error.message}`, messageType: 'error' });
            console.log(error)
        }
    }```

**Comment by stplaydog** [comment]
> Yo community, im using wagmi in my react staking dapp and now i wanna test it, but when i click on stake button it doesnt send metamask tx window, but in mainnet it send me tx window.
> 
> error: **TypeError: Cannot destructure property 'account' of 'parameters' as it is undefined.**
> 
> here code:
> 
> ```
>    const { writeContract, isSuccess: writeSuccess, isError: writeError, isPending: writePending } = useWriteContract()
>    const { data: stakeData, status } = useSimulateContract({
>         address: '0x027133f2521FBc163Fe7A95AB205384411084B12',
>         abi: NYXEcosystemStakingABI,
>         functionName: "stake",
>         args: [10*10**18],
>         stateMutability: 'nonpayable',
>         chainId: 97 //bsc testnet
>       });
> 
> 
>     const handleInputClick = (input) => {
>         setCurrentInput(input);
>     }
> 
>     const handleStake = async () => {
>         setTxPending(true);
>         try {
>             await writeContract(stakeData?.request)
>         } catch (error) {
>             setTxPending(false);
>             setMessageInfo({ isLoading: false, messageText: `Error: ${error.message}`, messageType: 'error' });
>             console.log(error)
>         }
>     }```
> ```

why this issue is closed?
